sched/distributor/poolDistributor_test.go

```diff
@@ -0,0 +1,68 @@
+package distributor
+
+import (
+	"github.com/golang/mock/gomock"
+	"github.com/scootdev/scoot/sched"
+	cm "github.com/scootdev/scoot/sched/clustermembership"
+	"testing"
+)
+
+func TestPoolDistributor_NilCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	clusterMock := cm.NewMockCluster(mockCtrl)
+	clusterMock.EXPECT().Members().Return(nil)
+
+	dist := NewPoolDistributor(clusterMock)
+
+	if nil != dist {
+		t.Error("Expected dist to be nil when no members in Cluster")
+	}
+}
+
+func TestPoolDistributor_EmptyCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	nodes := make([]cm.Node, 0)
+
+	clusterMock := cm.NewMockCluster(mockCtrl)
+	clusterMock.EXPECT().Members().Return(nodes)
+
+	dist := NewPoolDistributor(clusterMock)
+
+	if nil != dist {
+		t.Error("Expected dist to be nil when no members in Cluster")
+	}
+}
+
+func TestPoolDistributor_OneNodeCluster(t *testing.T) {
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	nodes := cm.GenerateTestNodes(1)
+	clusterMock := cm.NewMockCluster(mockCtrl)
```

Comment:
I agree, if the Interface fundamentally changes a lot of the test cases will break or need to be updated,  but I think this is the case even with a fake.  The calling code is only interacting with the interface so if the method it calls changes, the tests will need to be updated, which would also involve updating the fake.  I'd have to add the new method and have it return what I want.  

Also I'm fine with test cases being deleted and re-written if the implementation changes significantly.  I think this is healthy in an evolving code base, instead of trying to retrofit old test cases to support the modified code.  

I think you will see less Mock abuse in Go than other languages since you only can Mock interface, and not objects themselves.  Making a breaking change to an interface is going to cause a rather large refactor anyway including the Mocks.